JS API: Validate that importer result 'contents' is a `string` and improve ArgumentError output

Goodwine opened the pull request:
See: https://github.com/sass/sass-spec/pull/1834

Example output:
```
sass.Exception [Error]: Invalid argument (contents): must be a string but was: Buffer.: Instance of 'NativeUint8List'
  ╷
1 │ @import './foo.wxss';
  │         ^^^^^^^^^^^^
  ╵
  - 1:9  root stylesheet
  - [...]
```

Before using `toString()` for Argument error:
```
sass.Exception [Error]: must be a string but was: Buffer
  ╷
1 │ @import './foo.wxss';
  │         ^^^^^^^^^^^^
  ╵
  - 1:9  root stylesheet
  - [...]
```
Original error:
```
sass.Exception [Error]: NoSuchMethodError: method not found: 'charCodeAt' (receiver.charCodeAt is not a function)
  ╷
1 │ @import './foo.wxss';
  │         ^^^^^^^^^^^^
  ╵
  - 1:9  root stylesheet
  - [...]
```

Fixes https://github.com/sass/dart-sass/issues/1550

nex3 commented on CHANGELOG.md:
```diff
@@ -28,6 +28,11 @@
+### JS API
+
+* Importer results now validate whether `contents` is actually a string type.
+* Importer result argument errors are now rendered correctly.
```
Nit: We usually include a blank line between bullet points to help with plain-text readability.

nex3 reviewed (changes requested):
Can you add a test for this to `sass-spec` as well?

Goodwine commented:
> Can you add a test for this to sass-spec as well?

I wasn't sure where were the tests actually, but I found them now 🙂

Added tests and updated to also handle legacy importers which I realized just now via testing that it wasn't handled (yay tests)
